lib/file/bloc.dart

```diff
@@ -114,70 +116,66 @@ class FileBloc {
     // Let the user pick files.
     final files = await FilePicker.getMultiFile();
 
+    final progressUpdates = BehaviorSubject<UploadProgressUpdate>();
```

Comment:
For better separation of widget and service layer, I suggest you transform this into an `async*`-method returning a `Stream` of progress updates. The UI can then take care of displaying the `SnackBar`.